polkadot/node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -2111,3 +2032,99 @@ async fn handle_collation_fetch_response(
 	state.metrics.on_request(metrics_result);
 	result
 }
+
+// Returns how many seconded candidates and pending fetches are there within the view at a specific
+// relay parent
+fn seconded_and_pending_for_para_in_view(
+	implicit_view: &ImplicitView,
+	per_relay_parent: &HashMap<Hash, PerRelayParent>,
+	relay_parent: &Hash,
+	para_id: &ParaId,
+) -> usize {
+	// `known_allowed_relay_parents_under` returns all leaves within the view for the specified
+	// block hash including the block hash itself.
+	// If the relay parent is not in the view (unconnected candidate)
```

Comment:
Long story short - to handle unconnected candidates but this is unnecessary.

While testing I noticed that unconnected candidates are not counted correctly. I had multiple advertisements at the same relay parent and the seconded+pending count was always 0. 

However now I realise this should not happen in practice because once the `prospective-parachains` learns about such parent it will be in the view and on the next advertisement for it it will be returned by `known_allowed_relay_parents_under`. We will still get `None` on the first encounter but the count will be 0 so it's fine.

Bottom line - the behaviour of the `prospective-parachain` calls in the test was not correct and this `.or(...)` call is unnecessary.

Do you agree?

Reply:
unconnected candidates are candidates for which we don't yet have the full path to the latest included parablock.
but this refers to candidates, not relay chain blocks.

prospective parachains should get an active leaf update for the new parent before we get an advertisement on the new parent. At least this assumption is made throughout the codebase in multiple subsystems.

Reply:
Thank you for the explanation. I had a false understanding on this subject. I'm removing `.or(...)`